File: python/caesar_cipher.py
import string
def caesar_cipher(input_string, shift_amount):
    ans = ''
    special = set(string.punctuation)

    if shift_amount > 0:

        for char in input_string:
            if (char in special) or (char.isdigit()) or char == ' ':
                ans += char 
            
            elif char.isupper():
                for i in range(0, shift_amount, 1):
                    char = chr(ord(char) + 1)
                    if ord(char) > 90:
                        char = chr(65)
                ans += char 

            elif char.islower():
                for i in range (0, shift_amount, 1):
                    char = chr(ord(char) + 1)
                    if ord(char) > 122:
                        char = chr(97)
                ans += char 

    elif shift_amount < 0:
        for char in input_string: 
            if (char in special) or (char.isdigit()) or char == ' ':
                ans += char  
            elif char.isupper():
                for i in range(0, shift_amount, -1):
                    char = chr(ord(char) - 1)

                    if ord(char) < 65:
                        char = chr(90)
                ans += char 

            elif char.islower():
                for i in range (0, shift_amount, -1):
                    char = chr(ord(char) - 1)

                    if ord(char) < 97:
                        char = chr(122)
                ans += char         
    else:
        ans = input_string
    print(ans)
    return ans

File: js/caesarCipher.js
exports.caesarCipher = function(inputString, shiftAmount) {
    let ans = ''
    let temp = ''
    const special = "/[!@#$%^&*()1234567890 .]/"

    if(shiftAmount > 0){
        for (i = 0; i < inputString.length; i++){
            if (special.indexOf(inputString[i]) > -1){
                ans += inputString[i]
            }
            else if(inputString[i] == inputString[i].toUpperCase()){
                temp = inputString[i]
                for(j = 0; j < shiftAmount; j++){
                    temp = String.fromCharCode((temp.charCodeAt(0) + 1))
                    if (temp.charCodeAt(0) > 90){
                        temp = String.fromCharCode(65)
                    }
                }
                ans += temp
            }
            else{
                temp = inputString[i]
                for(j = 0; j < shiftAmount; j++){
                    
                    temp = String.fromCharCode((temp.charCodeAt(0) + 1))
                    if (temp.charCodeAt(0) > 122){
                        temp = String.fromCharCode(97)
                    }
                }
                ans += temp
            }
            
        }
        console.log(ans)
        return ans
    }

    else if (shiftAmount < 0){
        for (i = 0; i < inputString.length; i++){
            if (inputString[i] == ' ' || special.indexOf(inputString[i]) > -1){
                ans += inputString[i]
            }
            else if(inputString[i] == inputString[i].toUpperCase()){
                temp = inputString[i]
                for(j = 0; j > shiftAmount; j--){
                    temp = String.fromCharCode((temp.charCodeAt(0) - 1))
                    if (temp.charCodeAt(0) < 65){
                        temp = String.fromCharCode(90)
                    }
                    
                }
                ans += temp
            }
            else{
                temp = inputString[i]
                for(j = 0; j > shiftAmount; j--){
                    temp = String.fromCharCode((temp.charCodeAt(0) - 1))
                    if (temp.charCodeAt(0) < 97){
                        temp = String.fromCharCode(122)
                    }

                }
                ans += temp
            } 
        }
        console.log(ans)
        return ans
    }

    else{
        console.log(inputString)
        return inputString
    }
    
};
